datafusion/functions-aggregate/src/expr_builder.rs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use datafusion_expr::{expr::AggregateFunction, Expr};
+use sqlparser::ast::NullTreatment;
+
+/// Builder for creating an aggregate function expression
```

Comment:
I suggest:
1. Add a docstring example here
2. Consider putting this in `datafusion/expr` (not `datafusion-functions-aggregate`) -- if we put it in `datafusion-functions-aggregate` people can't use it unless they use the built in aggregates of DataFusion 🤔

Reply:
> if we put it in datafusion-functions-aggregate people can't use it unless they use the built in aggregates of DataFusion

Nice! I forgot this